test/elixir/test/mango/02_basic_find_test.exs

```diff
@@ -25,4 +25,300 @@ defmodule BasicFindTest do
+    Enum.each([0, 1, 5, 14], fn s ->
+      {:ok, docs} = MangoDatabase.find(@db_name, %{"age" => %{"$gt" => 0}}, skip: s)
+      assert length(docs) == (15 - s)
+    end)
+  end
+
+  test "sort" do
+    {:ok, docs1} = MangoDatabase.find(@db_name, %{"age" => %{"$gt" => 0}}, sort: [%{"age" => "asc"}])
+    docs2 = Enum.sort_by(docs1, fn d -> d["age"] end)
+    assert docs1 == docs2
+
+    {:ok, docs1} = MangoDatabase.find(@db_name, %{"age" => %{"$gt" => 0}}, sort: [%{"age" => "desc"}])
+    docs2 = Enum.reverse(Enum.sort_by(docs1, fn d -> d["age"] end))
+    assert docs1 == docs2
+  end
+
+  test "sort desc complex" do
+    {:ok, docs} = MangoDatabase.find(@db_name,
+      %{
+        "company" => %{"$lt" => "M"},
+        "$or" => [%{"company" => "Dreamia"}, %{"manager" => true}],
+      },
+      sort: [%{"company" => "desc"}, %{"manager" => "desc"}],
+    )
+    companies_returned = Enum.map(docs, fn doc -> doc["company"] end)
+    desc_companies = Enum.sort(companies_returned, :desc)
+    assert desc_companies == companies_returned
+  end
+
+  test "sort with primary sort not in selector" do
+    {:error, resp} = MangoDatabase.find(@db_name,
+      %{"name.last" => %{"$lt" => "M"}},
+      sort: [%{"name.first" => "desc"}]
+    )
+    assert resp.status_code == 400
+    assert resp.body["error"] == "no_usable_index"
+  end
+
+  test "sort exists true" do
+    {:ok, docs1} = MangoDatabase.find(@db_name,
+      %{"age" => %{"$gt" => 0, "$exists" => true}},
+      sort: [%{"age" => "asc"}]
+    )
+    docs2 = Enum.sort_by(docs1, fn d -> d["age"] end)
+    assert docs1 == docs2
+  end
+
+  test "sort desc complex error" do
+    {:error, resp} = MangoDatabase.find(@db_name,
+      %{
+        "company" => %{"$lt" => "M"},
+        "$or": [%{"company" => "Dreamia"}, %{"manager" => true}],
+      },
+      sort: [%{"company" => "desc"}],
+    )
+    assert resp.status_code == 400
+    assert resp.body["error"] == "no_usable_index"
+  end
+
+  test "fields" do
+    selector = %{"age" => %{"$gt" => 0}}
+    {:ok, docs} = MangoDatabase.find(@db_name, selector, fields: ["user_id", "location.address"])
+    Enum.each(docs, fn d ->
+      assert Enum.sort(Map.keys(d)) == ["location", "user_id"]
+      assert Enum.sort(Map.keys(d["location"])) == ["address"]
+    end)
+  end
+
+  test "r" do
+    Enum.each([1, 2, 3], fn r ->
+      {:ok, docs} = MangoDatabase.find(@db_name, %{"age" => %{"$gt" => 0}}, r: r)
+      assert length(docs) == 15
+    end)
+  end
+
+  test "empty" do
+    {:ok, docs} = MangoDatabase.find(@db_name, %{})
+    # 15 users
+    assert length(docs) == 15
+  end
+
+  test "empty subsel" do
+    {:ok, docs} = MangoDatabase.find(@db_name, %{"_id" => %{"$gt" => nil}, "location" => %{}})
+    assert length(docs) == 0
+  end
+
+  test "empty subsel match" do
+    resp = MangoDatabase.save_docs(@db_name, [%{"user_id" => "eo", "empty_obj" => %{}}])
+    {:ok, docs} = MangoDatabase.find(@db_name, %{"_id" => %{"$ne" => nil}, "empty_obj" => %{}})
```

Comment:
in the python is `gt` but here is `ne`, is there a reason?

Reply:
same as above ☝️